Code/Framework/AzCore/AzCore/std/containers/span.h

```diff
@@ -112,9 +112,9 @@ namespace AZStd
         template<size_t N, class = enable_if_t<extent == dynamic_extent || N == Extent>>
         constexpr span(type_identity_t<element_type> (&arr)[N]) noexcept;
 
-        template <class U, size_t N, class = enable_if_t<extent == dynamic_extent || N == Extent>>
+        template <class U, size_t N, class = enable_if_t<(extent == dynamic_extent || N == Extent) && is_convertible_v<U, T>>>
         constexpr span(array<U, N>& data) noexcept;
-        template <class U, size_t N, class = enable_if_t<extent == dynamic_extent || N == Extent>>
+        template <class U, size_t N, class = enable_if_t<(extent == dynamic_extent || N == Extent) && is_convertible_v<U, T>>>
```

Comment:
This makes sense, as this constraints the template parameter `U` to be convertible to the span type `T`.

It was unintended that the previous code accepted any type for`U` so this fix is good.

Before an `AZStd::span<int>` could accept an `AZStd::array<*>` which succeed at adding the AZStd::array constructor to the template function candidate set, but then it would fail when it parsed the body of the function.
This change makes it so that parameter is constrained to  `AZStd::array<ConvertibleToIntType>`